File: problem.set.1/problem.3.'CCC '14 S1 - Party Invitation'.py
k = int(input())
m = int(input())

friends = [i for i in range(1, k+1)] # make a list of friends from 1 to k
removal = []

for i in range(m):
    removal.append(int(input())) # list of m's

for i in removal: # each i is an m in the list of removal
    x = 1
    for j in range(1, len(friends) // i + 1): # multiples of i * j for integer n = [1, len(friends) // i + 1]
        # e.g. if i = 2 and len(friends) = 10 --> i * j = 2, 4, 6, 8, 10
        friends.pop(i * j - x) # need to subtract a value b/c we're popping, index is shifting to left each time
        x += 1 # every pop needs to shift left 1 more time

for i in friends:
    print(i)

File: problem.set.1/problem.3.'CCC '14 S1 - Party Invitation'.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input the total number of friends and number of removal rounds
        int k = scanner.nextInt();
        int m = scanner.nextInt();

        // Create a list of friends from 1 to k
        ArrayList<Integer> friends = new ArrayList<>();
        for (int i = 1; i <= k; i++) {
            friends.add(i);
        }

        // Create a list of m removal numbers
        ArrayList<Integer> removal = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            removal.add(scanner.nextInt());
        }

        // Remove friends based on the multiples in the removal list
        for (int i : removal) {
            int x = 0; // counter for shifting indices after each pop
            for (int j = 1; j * i - x <= friends.size(); j++) {
                friends.remove(i * j - 1 - x); // pop the friend at i * j, adjust index due to previous pops
                x++;
            }
        }

        // Print the remaining friends
        for (int friend : friends) {
            System.out.println(friend);
        }

        scanner.close();
    }
}